decisions/2024-04-15_decimal_pack_size.md

```diff
@@ -0,0 +1,131 @@
+- Should be palatable for integrations with other systems
+
+### Additional Context
+
+- mSupply back and forth trying to make decimals calculate/present correctly
+  - [Tax rate](https://github.com/msupply-foundation/msupply/issues/10118#event-12127072697)
+  - [ledger](https://github.com/msupply-foundation/msupply/issues/10508)
+- [HSH KDD](https://github.com/msupply-foundation/health-supply-hub/blob/develop/docs/decisions/002-storing-monetary-values.md) generally deciding to use floats for monetary values.
+- [HSH issue](https://github.com/msupply-foundation/health-supply-hub/issues/637) deciding that quantity, pack_size and unit_quantity needs to be float for easiest interoperability with mSupply tenders and consistency. Several changes were added to mSupply to strongly deter uses from using decimal pack sizes in their tender lines and making them confirm it.
+- [OMS discussion](https://github.com/msupply-foundation/open-msupply/discussions/2905) for decimal values in stocktakes.
+- [Internal OMS discussion](https://github.com/msupply-foundation/open-msupply/discussions/29) for decimal values in ledger and UI.
+- [SAGE](https://www.greytrix.com/blogs/sageaccpacerp/2015/01/07/allow-fractional-quantities-in-purchase-order/) has a preference - allows 4 decimal places if on, zero if off. Permanent if switched on.
+- [SAP](https://content.cnbssoftware.com/blog/rounding-issues-sap-uom) blog post detailing that SAP allows decimals. In this case appears to be 3dp allowed. They have Unit of Measure of "case", which contains 6 units of "each" (aka a box of 6 bottles), but it can be split as sometimes is required IRL. So 1/6 = 0.167. 6\*0.167 = 1.002 though ;-). Shenanigans are had - if stock out of 1 "each" then they make sure the source keeps the remainder i.e. `1 case - 1 each = 0.834` stock left at the source, rather than `0.833`. The receiving location gets `0.166`, rather than `1.67`. This way the amount in stock at the source location is never a partial each (>0, but <1). But now the receiver location has a partial each! When they do the next sales order, they bump the quantity in the order up to account for the "fractional dust" but are careful to only do it within a threshold to make sure it doesn't appear to send/create stock that never existed...
+
+## Options
+
+### Option 1 - Support Decimal Pack Size like Legacy mSupply
+
+This is the most straight forward for migration/integration as it's 1-to-1, the fields are simply the same numeric type. Downsides include introducing another avenue of floating point inaccuracy in arithmetic. It makes sense to users, as they entered the data and it is presented in the manner they entered it.
+
+There is potential for discrepancies to occur, e.g. store A repacks `1 * 1L` to `3 * 0.33L`. There are 0.01 packs of the original line... do we just zero it? If we issue all 3 to another store, they receive 0.99L. Do we round it up? What's the threshold?
```

Comment:
I would have though that in the context of this discussion, repeating decimal cannot happen, number of packs can be repeating decimal, but I don't see how pack size can be entered to be repeating decimal ?

Reply:
To use the SAP example they might have 6 "each" in a "case". If you tried to repack and manage stock as such... 
![image](https://github.com/msupply-foundation/open-msupply/assets/7684221/ea426d12-08e4-49a4-bd13-b73c4e8f4cfa)

I think somewhat contrived as I'm not aware of anyone doing that in mSupply, but technically nothing stopping you. Also clearly in the wild this sort of thing can happen given that SAP article.